### ISSUE TITLE
Reorganize package

### ISSUE THREAD

**Issue by tsalo** [FIU-Neuro/brainconn#19]
@adamkimbler Were you interested in reorganizing the submodules in the library? At the very least, I want to eliminate the `from X import *` patterns in the `__init__` files which give access to all of the functions in the library at the top level. Personally, I think that crowds the top-level namespace too much.

**Comment by tsalo** [comment]
I was thinking we could eliminate `algorithms` and just move all of its submodules up a level (e.g., `brainconn.algorithms.centrality` -> `brainconn.centrality`). WDYT?

We should probably also compile a comprehensive list of the functions we want to consolidate (e.g., `brainconn.centrality.betweenness_wei` and `brainconn.centrality.betweenness_bin` --> `brainconn.centrality.betweenness`).

**Comment by adamkimbler** [comment]
I agree, the algorithms just makes the namespace overall very clunky. Your examples are exactly how I believe the namespace should work. I have a list of the functions for consolidation that I can upload later.

**Comment by tsalo** [comment]
Awesome! If you add them as a [checklist](https://help.github.com/articles/about-task-lists/#creating-task-lists) we can quantify our progress on them and link this to a milestone (I'm thinking [0.0.2](https://github.com/FIU-Neuro/brainconn/milestone/2)).

**Comment by tsalo** [comment]
Also, we should agree on a general size limit to files, which will influence how we organize the library. In other projects, we've agreed to a limit of around 500 lines of code, ignoring the docstrings. I think that that is a reasonable length for users to digest, but it will mean splitting some files, like `brainconn/algorithms/reference.py`

**Comment by tsalo** [comment]
I've started work on this in #25. My changes eliminate the `algorithms` submodules and create new `centrality`, `clustering`, `core`, `degree`, `distance`, `generative`, `modularity`, `motifs`, `physical_connectivity`, `reference`, and `similarity`.
